### DISCUSSION TITLE
Query on dynamic wavelength allocation.

### DISCUSSION THREAD

**Discussion by oolongkiller** [Telecominfraproject/oopt-gnpy#479] [Q&A] [answered]
Hi,
Is it possible for Gnpy to realize dynamic wavelength allocation?

For example, I want to send wavelength cluster(maybe 1500-1525nm) at site-a and send another cluster(maybe 1525-1550nm) at site-b.  I used to think that I could use "roadm"  to cope with this situation. However, "roadm" in Gnpy doesn't seem to have the so-called add-and-drop function.

Can you give me some advice or  tell me the specific function of "roadm" in Gnpy?

Thankyou

**Comment by EstherLerouzic** [comment]
Dear @oolongkiller 
Could you please elaborate a bit more on your need ?
My understanding is that you would like to check feasibility on a path from Roadm A to Roadm B for a set of channels and on a second path from Roadm A to Roadm C for another set of channels. 

If this is the case then I suggest to have two separate simulations with gnpy-transmission-example with different spectrum (-spectrum option) or to use the planning gnpy-path-request script with a list of two services with different transceivers frequency range.

hope this helps,

regards

Esther

**Reply by oolongkiller** [reply]
Dear @EstherLerouzic 

Thanks for your answer.

Sorry, my expression was not clear enough.

There is now a situation where two transmitters are located at Site-A and Site-B, one receiver is located at Site-C, and Site-B exists on the path from Site-A to Site-C. I hope to send a wavelength cluster(1500-1525nm) from Site-A to Site-C. When the transmission passes through Site-B, Site-B sends another wavelength cluster(1525-1550nm) and automatically adds it to the transmission path, and finally detects the entire channel(include 1500-1550nm) parameters at Site-C. Can this situation be achieved?

Thanks

**Reply by EstherLerouzic** [reply, marked as answer]
Dear @oolongkiller ,
Although the GNPy library permits this, I think that the current scripts do not allow to set this type of setup. Both assume that the propagation of all channels is on the same path. You would need to create your own script to do that.

regards
Esther
Esther

**Reply by oolongkiller** [reply]
Many thanks for your reply!

**Reply by oolongkiller** [reply]
Dear @EstherLerouzic 

Sorry to bother you again.

Does "roadm" only support equalizing channels? If it is possible to add-and-drop fixed wavelengths that I have set up. 

Another question is how to realize a non-uniform spectral information input. I have seen the info.py which includes the definition "create_arbitrary_spectral_information", but I still have no idea to use it. Can you give me some advice?

Thanks

**Reply by jktjkt** [reply]
GNPy (as in the SW that you can download and use without any programming) assumes that all your links have the same channel load. This channel load is used for planning and network optimization (because you don't know up front how things are going to look like in five years). When it comes to actual propagation, it's possible to run the simulation and ask for results betwen A and B, or B and C, or A and C, or any or all of them. In this setup, the Add/Drop part of each ROADM is "implied"; one doesn't have to describe it, it's "always there" (and it's reflected in the GSNR of the resulting signal based on the estimated OSNR impact of the A/D stage). In the topology it's represented by transponders connecting "directly" to the ROADM.

Flex spectrum setup is [described here in the docs](https://gnpy.readthedocs.io/en/master/json.html#arbitrary-channel-definition).

If you are a SW engineer (or have SW engineers at your disposal), you can use the Python SW library that is at the core of GNPy and ask it do arbitrary simulations.

**Reply by oolongkiller** [reply]
Dear @jktjkt 

Many thanks for your reply.

Can you elaborate a bit more on the "implied Add/Drop part"? Does it means that I cannot choose the wavelength I expect to add/drop?

Actually, I want to realize the situation where some wavelengths(λ1, λ2, λ3, λ4) are transmitted from A to C, and when passing through the ROADM, a wavelength(λ3) is dropped and a wavelength(λ5) is added. Is it possible to equivalently simulate the above scenario through two simulations, namely the first transmission of wavelengths(λ1, λ2, λ4) from A to C  and the second transmission of wavelength(λ5) from ROADM to C.

Thanks
